overlord/snapstate/target.go

```diff
@@ -344,6 +344,24 @@ func checkTargetAgainstValidationSets(target target, action string, vsets *snapa
 	return checkComponentsAgainstConstraints(target.info.SnapName(), comps, constraints, action)
 }
 
+func checkSnapActionAgainstValidationSets(sar store.SnapActionResult, components []ComponentSetup, action string, vsets *snapasserts.ValidationSets) error {
```

Comment:
couldn't this and the Target variant of the function share more code in a helper maybe taking a snap.Info and []ComponentSetup ?